gravitational/teleport
# Added IAM Authentication with IAM role support for Amazon Redshift

Closes #27383

changelog: Added IAM Authentication with IAM role support for Amazon Redshift

TODO:

- [x] Updated configurator
- [x] Added UT
- [x] Added Doc (#34940) ([rendered](https://docs-6imgtwas5-goteleport.vercel.app/docs/database-access/guides/postgres-redshift/#step-35-create-an-iam-policy-for-teleport))

Setup: 
1. Make a new IAM role, make sure it trusts the IAM role of the database service
2. Add permission to the above IAM role (make sure to adjust to your setup by replacing the account-id and cluster-id)
```code
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Effect": "Allow",
            "Action": "redshift:GetClusterCredentialsWithIAM",
            "Resource": "arn:aws:redshift:ca-central-1:{account-id}:dbname:{cluster-id}/*"
        }
    ]
}
```
3. Setup a Redshift cluster and add it to Teleport

Example:
```bash
$ tsh db connect --db-user role/steve-poweruser --db-name test2 steve-redshift
```
